buildSrc/src/main/java/com/amazonaws/blox/tasks/GenerateSwaggerModel.java

```diff
@@ -76,18 +77,23 @@
    */
   @Nested private List<SwaggerFilter> filters = new ArrayList<>();
 
-  public GenerateSwaggerModel() {
-    filters.add(DEFAULT_FILTER);
-  }
-
   @TaskAction
-  public void generateSpec() throws IOException, ClassNotFoundException, GenerateException {
+  public void generateSpec() throws IOException, GenerateException {
     ClassLoader loader = projectClassLoader();
-
     Set<Class<?>> classes = new HashSet<>();
 
-    for (String name : apiClasses) {
-      classes.add(loader.loadClass(name));
+    for (String pkg : apiPackages) {
+      Reflections reflections =
+          new Reflections(
+              new ConfigurationBuilder()
+                  .addClassLoaders(loader)
+                  .addUrls(ClasspathHelper.forPackage(pkg, loader))
+                  .setExpandSuperTypes(false)
+                  .filterInputsBy(new FilterBuilder().includePackage(pkg)));
+      Set<Class<?>> springMVCClasses =
+          reflections.getTypesAnnotatedWith(
+              org.springframework.web.bind.annotation.RestController.class);
```

Comment:
Hm, we don't have a direct dependency on this in the `build.gradle`, is it coming in as a transitive dependency? Might be worth adding an explicit dependency in that case.

Reply:
Yes, it comes from the "swagger-maven-plugin", which we use to generate swagger from Spring mvc.  

Not sure what's the value we add for explicit dependency for spring-web. I would see it's better to use the transitive dependency from swagger to avoid potential problem to maintain the same spring version within swagger or different spring versions.